packages/tx/src/legacyTransaction.ts

```diff
@@ -76,6 +77,20 @@ export default class Transaction extends BaseTransaction<Transaction> {
 
     const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = values
 
+    const integerProps = [nonce, gasPrice, gasLimit, value]
+
+    for (const x of [v, r, s]) {
+      // If transaction is signed, add v, r, and s to values checked for leading zeroes
+      if (x?.length) integerProps.push(x)
+    }
```

Comment:
I guess I would feel better here if we rather do a `values.length === 9` check for a signed tx (as being done above) and *then* just add all values for a check. Feels more consistent to me than this one by one selection.